global/global.go

```diff
@@ -139,13 +145,25 @@ func (t *TLS) Config() (*tls.Config, error) {
 	caCertPool := x509.NewCertPool()
 	caCertPool.AppendCertsFromPEM(cert)
 
+	// only have CA file, go TLS
+	if len(t.Cert) <= 0 || len(t.Key) <= 0 {
+		log.Debug("[TLS] Only have CA file, go TLS")
+		return &tls.Config{
+			RootCAs:            caCertPool,
```

Comment:
x509.NewCertPool creates a new empty cert pool.  
x509.SystemCertPool returns a copy of the system cert pool.

I think NewCertPool will override the default system CA configurations, which would make the legitimate certs signed by public authorities verify failed. Is it your expectation?

Reply:
sorry, I merge this pr so quickly.

i think the better way should be 

```diff
@@ -142,7 +142,10 @@ func (t *TLS) Config() (*tls.Config, error) {
        if err != nil {
                return nil, err
        }
-       caCertPool := x509.NewCertPool()
+       caCertPool, _ := x509.SystemCertPool()
+       if caCertPool == nil {
+               caCertPool = x509.NewCertPool()
+       }
        caCertPool.AppendCertsFromPEM(cert)
 
        // only have CA file, go TLS
```

I will open another PR to address this.

Reply:
if cert then create new pool
else dothing and goruntime will use system pool

should not call add to system pool